src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -0,0 +1,255 @@
+package edu.kit.ipd.crowdcontrol.objectservice.moneytransfer;
+
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.GiftCodeRecord;
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.WorkerRecord;
+import edu.kit.ipd.crowdcontrol.objectservice.database.operations.WorkerBalanceOperations;
+import edu.kit.ipd.crowdcontrol.objectservice.database.operations.WorkerOperations;
+import edu.kit.ipd.crowdcontrol.objectservice.mail.MailHandler;
+import edu.kit.ipd.crowdcontrol.objectservice.template.Template;
+
+import org.jooq.Result;
+
+import java.io.*;
+
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Describes a Manager for money transfers. The Manager can log payments and pay off.
+ *
+ * @author Felix Rittler
+ */
+public class MoneyTransferManager {
+
+    private MailHandler mailHandler;
+    private WorkerBalanceOperations workerBalanceOperations;
+    private WorkerOperations workerOperations;
+
+    private int payOffThreshold;
+    private int minGiftCodesCount;
+
+    private String notificationMailAddress;
+    private StringBuilder notificationText;
+
+    private ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    private ScheduledFuture<?> schedule = null;
+
+    private static final Logger LOGGER = LogManager.getRootLogger();
```

Comment:
Should be: `private static final Logger LOGGER = LogManager.getLogger(MoneyTransferManager.class);`